### ISSUE TITLE
shell: directive is buggy...

### ISSUE THREAD

**Issue by mmlb** [jwilm/alacritty#281]
I've been trying to get alacritty to run many different programs and most (all?) have some buggy behavior. Here's what the relevant section of the config:
```
shell: /home/manny/bin/alacrity-shell
```
_(it'd be great for alacrity to expand ~ here)_

And the contents of alacrity-shell have been trying to start tmux, nvim (with term://$SHELL), and $SHELL to some sort of woe. The worst has been nvim which for some reason can sometimes lock up my system quite badly.

I'll be calling alacritty from pangoterm for all the examples.
First up is tmux. I have my tmux.conf setup to create a new session so that `alacritty-shell` can just be `tmux attach`

```
❯ cat bin/alacritty-shell 
#!/usr/bin/env bash

tmux attach -t default
#zsh --login -i
#$SHELL --login -i
#nvim term://zsh

```

```
new-session -s default
set -g prefix C-s
unbind C-b
...
```

```
❯ killall tmux
tmux: no process found
                                                                                                                  
~  
❯ alacritty
device_pixel_ratio: 1
width: 1144, height: 1394
Initializing glyph cache
num fonts = 244
num fonts = 244
num fonts = 244
Finished initializing glyph cache in 0.071625394
set_inner_size: 804px × 436px
width: 804, height: 436
Cell Size: (10 x 18)
PTY Dimensions: Line(24) x Column(80)
num_cols, num_lines = 114, 77
width: 1144, height: 1394
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[Unhandled CSI] action='l', args=[1005], intermediates=[63]
[unhandled osc_start]
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='2'
[unhandled osc_end] byte=0
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled osc_start]
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='2'
[unhandled osc_end] byte=0
[Unhandled CSI] action='l', args=[1005], intermediates=[63]
thread 'pty reader' panicked at 'unexpected read err: Error { repr: Os { code: 5, message: "Input/output error" } 
}', src/event_loop.rs:227
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Goodbye
                                                                                                                  
~  
❯ bin/alacritty-shell 
[exited]
```
so alacritty exits/fails, but when I called alacritty-shell manually I was able to exit cleanly. Now, if I run tmux and detach, alacritty works just fine.

```
~  
❯ bin/alacritty-shell
[detached (from session default)]
                                                                                                                  
~ 1529ms 
❯ alacritty          
device_pixel_ratio: 1
width: 1144, height: 1394
Initializing glyph cache
num fonts = 244
num fonts = 244
num fonts = 244
Finished initializing glyph cache in 0.072823565
set_inner_size: 804px × 436px
width: 804, height: 436
Cell Size: (10 x 18)
PTY Dimensions: Line(24) x Column(80)
num_cols, num_lines = 114, 77
width: 1144, height: 1394
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[Unhandled CSI] action='l', args=[1005], intermediates=[63]
[unhandled osc_start]
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='2'
[unhandled osc_end] byte=0
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
[unhandled osc_start]
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='1'
[unhandled osc_put] byte='2'
[unhandled osc_end] byte=0
[Unhandled CSI] action='l', args=[1005], intermediates=[63]
Goodbye
```
 I'm having a worse time with nvim (my preferred multiplexer), but figured I'd go with tmux since that should work only to find this situation.

When I try with the nvim line I get:
```
❯ alacritty
device_pixel_ratio: 1
width: 1144, height: 1394
Initializing glyph cache
num fonts = 244
num fonts = 244
num fonts = 244
Finished initializing glyph cache in 0.072128687
set_inner_size: 804px × 436px
width: 804, height: 436
Cell Size: (10 x 18)
PTY Dimensions: Line(24) x Column(80)
num_cols, num_lines = 114, 77
width: 1144, height: 1394
[Unhandled CSI] action='h', args=[1004], intermediates=[63]
[Unhandled CSI] action='t', args=[8, 77, 114], intermediates=[]
[unhandled] esc_dispatch params=[], ints=[40], byte='B' (42)
^C
```

**Comment by mmlb** [comment]
So I just noticed that alacritty sets `$SHELL` with whatever is in the `shell: ` directive and that's what was messing up my `$SHELL --login -i` attempt along with my (not shown here) `nvim term://$SHELL`. I don't think alacritty should be setting the $SHELL var...

For the `$SHELL --login -i` case bash comes back with a recursion error message:
```
bash : warning: shell level (1000) too high, resetting to 1
bash : warning: shell level (1000) too high, resetting to 1
bash : warning: shell level (1000) too high, resetting to 1
...
```

**Comment by mmlb** [comment]
so looks like 2 possibly 3 bugs...
$SHELL
plain `nvim term://zsh`
and w/e is going on with `tmux`...
Should I create separate issues?

**Comment by jwilm** [comment]
I think just having the one issue is fine. We can fix all of these problems at once.

I don't quite understand the 2nd/3rd bugs - it sounds like not setting $SHELL may fix everything?

**Comment by mmlb** [comment]
not setting shell should fix the case of when `$SHELL` is used within `shell` binary/script. My original file had
```
nvim term://$SHELL
```
and that was horrible, probably because of the interaction of `$SHELL` *and* nvim.

```
nvim term://zsh
```
still does not work but it may be due to something in neovim reading `$SHELL` internally (I'll check my nvimrc file and see if I can find anything in nvim docs or code)

Thanks for alacritty btw this looks great!

**Comment by NickeZ** [comment]
I think this should be fixed since 5b079f6d437c567a1533a719194f6e579a7eedae

**Comment by jwilm** [comment]
Agree @NickeZ; thanks!
